java/algorithm/src/test/java/org/apache/arrow/algorithm/sort/TestInsertionSorter.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.algorithm.sort;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.IntVector;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Test cases for {@link InsertionSorter}.
+ */
+public class TestInsertionSorter {
+
+  private BufferAllocator allocator;
+
+  @Before
+  public void prepare() {
+    allocator = new RootAllocator(1024 * 1024);
+  }
+
+  @After
+  public void shutdown() {
+    allocator.close();
+  }
+
+  private void testSortIntVectorRange(int start, int end, int vectorLength) {
+    try (IntVector vector = new IntVector("vector", allocator);
+         IntVector buffer = new IntVector("buffer", allocator)) {
+
+      vector.allocateNew(vectorLength);
```

Comment:
please use vector populator instead of loops.

Reply:
Revised. Thanks for the good point.

Reply:
This wasn't what I had in mind.  It is still hard to understand what the tests are doing.

Creating a Vector of [5, 4, 3, 2, 1]  then showing that insert for the middle three elements is [5, 2, 3, 4, 1] is much more helpful (if longer arrays are needed to demonstrate then that is fine.  I don't think an array larger then 10 should be needed?  

Google has published some of their internal [tips](https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html) on testing, I think the one linked describes my thinking on this.  The others are worth reading as well.

Reply:
@emkornfield Thanks for the article and good suggestion. 
I have revised the tests accordingly. Please take a look.